lib/jemoji.rb

```diff
@@ -8,8 +8,16 @@ class Emoji
 
     class << self
       def emojify(doc)
+        return unless doc.output =~ HTML::Pipeline::EmojiFilter.emoji_pattern
         src = emoji_src(doc.site.config)
-        doc.output = filter_with_emoji(src).call(doc.output)[:output].to_s
+        if doc.output =~ /<body/
+          parsed_doc = Nokogiri::HTML::Document.parse(doc.output)
+          body       = parsed_doc.at_css('body')
+          body.replace filter_with_emoji(src).call(body.to_html)[:output]
```

Comment:
I think we're going to have to pass it as a string... `body` at that point is a `Nokogiri::XML::Element`, not a document, and thus has no `parent` method (and errors out on `has_ancestor?`).